glue/qt/widgets/histogram_widget.py

```diff
@@ -193,6 +183,21 @@ def component(self, component):
 
     @defer_draw
     def _set_attribute_from_combo(self, *args):
+        if self.component is not None:
+            for d in self._data:
+                try:
+                    component = d.get_component(self.component)
+                except:
+                    continue
+                else:
+                    break
```

Comment:
There's no directly link from `CompoentID` to `Component`, so some loop like this seems unavoidable to me. In general a ComponentID can belong to >1 data objects in a DataCollection. I could se the argument for making this a utility method of DataCollection.